orbit/pkg/osservice/osservice_windows.go

```diff
@@ -0,0 +1,112 @@
+//go:build windows
+// +build windows
+
+package osservice
+
+import (
+	"errors"
+	"os"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/orbit/pkg/constant"
+	"github.com/fleetdm/fleet/v4/orbit/pkg/platform"
+
+	"github.com/rs/zerolog/log"
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc"
+)
+
+type windowsService struct {
+	shutdownFunctions   *[]func(err error)
+	fleetDesktopPresent bool
+}
+
+func (m *windowsService) bestEffortShutdown() {
+	serviceShutdown := errors.New("service is shutting down")
+
+	// Calling interrupt functions to gracefully shutdown runners
+	for _, interruptFn := range *m.shutdownFunctions {
+		interruptFn(serviceShutdown)
+	}
+
+	// Now ensuring that no child process are left
```

Comment:
Now that you are interrupting Orbit, do we need the killing of processes here? (Orbit kills them when interrupted).
- https://github.com/fleetdm/fleet/blob/main/orbit/cmd/orbit/orbit.go#L702-L711
- https://github.com/fleetdm/fleet/blob/main//orbit/pkg/osquery/osquery.go#L180-L186

Reply:
Good question. I'm leaving the process-killing logic on service teardown to play safe and ensure that no process is left after interrupt is called. The logic to kill osqueryd is redundant as osqueryd interrupt() works well. Now on the other side, I found through testing that desktop-app interrupt() is failing to kill the desktop app. So the logic to force-kill the app is needed until I find what's going on with this interrupt() logic

Reply:
+1, better safe.